api/src/main/java/io/minio/MinioClient.java

```diff
@@ -1374,6 +1390,46 @@ public ObjectStat statObject(String bucketName, String objectName)
     return objectStat;
   }
 
+  /**
+   * Returns meta data information of given object in given bucket.
+   *
+   * </p><b>Example:</b><br>
+   * <pre>{@code ObjectStat objectStat = minioClient.statObject("my-bucketname", "my-objectname");
+   * System.out.println(objectStat); }</pre>
+   *
+   * @param bucketName Bucket name.
+   * @param objectName Object name in the bucket.
+   * @param sse        Encryption metadata
+   *
+   * @return Populated object metadata.
+   *
+   * @throws InvalidBucketNameException  upon invalid bucket name is given
+   * @throws NoResponseException         upon no response from server
+   * @throws IOException                 upon connection error
+   * @throws XmlPullParserException      upon parsing response xml
+   * @throws ErrorResponseException      upon unsuccessful execution
+   * @throws InternalException           upon internal library error
+   * @see ObjectStat
+   */
+  public ObjectStat statObject(String bucketName, String objectName, ServerSideEncryption sse)
+    throws InvalidBucketNameException, NoSuchAlgorithmException, InsufficientDataException, IOException,
+           InvalidKeyException, NoResponseException, XmlPullParserException, ErrorResponseException,
+           InternalException, InvalidArgumentException {
+    if ((sse.getType().toString().equals("SSE_S3")) 
+        || (sse.getType().toString().equals("SSE_KMS"))) {
+      throw new InvalidArgumentException("Encryption type is not supported for this operation.");
+    } else if ((sse.getType().toString().equals("SSE_C")) && (!this.baseUrl.isHttps())) {
```

Comment:
You can use `sse.getType() == ServerSideEnccryption.SSE_C` since enum. No need to string conversion.